### ISSUE TITLE
Never works on Raspberry Pi 3 on Arch Linux ARM

### ISSUE THREAD

**Issue by Dadoum** [checkra1n/BugTracker#790]
**Tell us about your setup:**
1. What iDevice are you using? iPhone 7 (Global)
2. On what version of iOS is it? iOS 12.4
3. What version of checkra1n are you using? GUI or CLI? 1.9.8 CLI
4. What is your host system (OS version? Hackintosh? VM? etc.)? Arch Linux ARM for RPi 3B+
5. How are you connecting to the device (USB-A? USB-C? Apple/3rd party cable? Through a USB hub?)? 3rd party cable, that works on my (Manjaro XFCE) computer.

**What are the steps to reproduce the issue?**
Normally just running it should show the issue, but just in case I tell the exact way I trigger:
1. Start `checkra1n -c -V` from local SSH to the RPI 3
2. See the USBDEVFS_SUBMITURB: No such device and checkra1n failing
...

**What do you expect, and what is happening instead?**
Jailbreak but it boots normally instead

**Does the issue also occur if you enable Safe Mode?**
Yes

**Any other info, error logs, screenshots, ...?**
Nope

**PS:** This issue is here because it never works, whereas it says on checkra1n website that it is unreliable. This can be due to Arch Linux installation that is very different to Raspbian, with totally different drivers.

**Comment by TheTiEr** [comment]
it is likely you are missing some drivers for the ios device.
install thre packages:  

ideviceinstaller python-imobiledevice libimobiledevice-utils python-plist ifuse usbmuxd libusbmuxd-tools

and try it again

**Comment by Dadoum** [comment]
The packages should be installed, but I cannot check exactly because **you name the package with their APT names instead of ones that can be found on AUR**

**Comment by InnaCL** [comment]
I have raspbian and I receive the same USBDEVFS_SUBMITURB error: No such device and checkra1n failing, and I have installed what is indicated in the apt packages

**Comment by MJavad** [comment]
I also have the same issue, with NanoPi K2 running Armbian and arm64 version
```
#
# Checkra1n beta 0.9.8
#
# Proudly written in nano
# (c) 2019-2020 Kim Jong Cracks
#
#========  Made by  =======
# argp, axi0mx, danyl931, jaywalker, kirb, littlelailo, nitoTV
# never_released, nullpixel, pimskeks, qwertyoruiop, sbingner, siguza
#======== Thanks to =======                                                                                                                                                           # haifisch, jndok, jonseals, xerub, lilstevie, psychotea, sferrini 
# Cellebrite (ih8sn0w, cjori, ronyrus et al.)                                                                                                                                         #==========================
- [*]: Waiting for DFU devices
- [*]: Exploiting
- [*]: Checking if device is ready 
- [*]: Setting up the exploit (this is the heap spray)
USBDEVFS_SUBMITURB: No such device
USBDEVFS_REAPURB: No such device
- [!]: Exploit Failed (error code: -28)
- [!]: Unknown error (error code: -28) 
- [!]: Exploit Failed (error code: -21)
- [!]: Unknown error (error code: -21) 
- [!]: Exploit Failed (error code: -21) 
- [!]: Unknown error (error code: -21) 
```

**Comment by MJavad** [comment]
I tried with a NanoPi Neo2 and it worked fine. It seems the problem with NanoPi K2 (and probably RPi 3) is that it's ports are from an onboard hub. Any idea how to work around that?

**Comment by IMS212** [comment]
This is a known issue, and yes it is caused by the rpi3 and lower’s hub. The rpi4 does not have this issue.

**Comment by sc7171** [comment]
This issue occurs for me also, I've tried Raspbian Buster Lite and Raspbian Buster Desktop on the **Raspberry Pi 3**, and still, I get the same issue with it failing.
ARGS: -c
_Note: I even installed the packages listed above, "ideviceinstaller python-imobiledevice libimobiledevice-utils python-plist ifuse usbmuxd libusbmuxd-tools", but it still fails._
Attached are my logs:

> Waiting for DFU devices
> Exploiting
> Checking if device is ready
> Setting up the exploit (this is the heap spray)
>USBDEVFS_SUBMITURB: No such device
>USBDEVFS_REAPURB: No such device
 > Exploit Failed (error code: -28)
 > Unknown error (error code: -28)
 > Exploit Failed (error code: -21)
 > Unknown error (error code: -21)
 > Exploit Failed (error code: -21)
 > Unknown error (error code: -21)

**Comment by ceedveed** [comment]
I have a different device (ARM based chromebook using crouton), but I get the exact same log. A x86_64 xubuntu laptop works though (need to set verbose boot).

**Comment by Skye7171** [comment]
This seems like a ARM issue. This problem is occurring on different hardware and operating systems that share the ARM architecture. The devs need to be aware of this as it is being confused with **ONLY the RASPBERRY PI 3**.

**Comment by MJavad** [comment]
@IMS212 Do you know where the issue is being tracked?

**Comment by ann0see** [comment]
Do we know if all the non working devices share the same USB controller?
What I tried so far:

Checkra1n doesn't run on my Raspberry Pi Zero W V1.1, Raspberry Pi 3 V1.2, Raspberry Pi Zero V1.3
But it does run on my Banana Pi M1 on Armbian. 
Did anybody try e.g. the Giant Board https://www.crowdsupply.com/groboards/giant-board Banana Pi M2 Zero etc?

I also wanted to link https://www.raspberrypi.org/documentation/hardware/raspberrypi/usb/README.md

Edit: As far as I understand correctly, the USB C Port of the Pi 4 is directly connected to the Processor but the USB A ports are connected to the USB Controller (which is later connected over PCIe to the Processor). So the new connection method of the Pi 4 is more like on a PC and therefore on a Pi 4 everything works fine.

So does this problem also exist if we plugin an iPhone to the Pi 4 via USB C (and power the Pi differently)?

Just looked into the Nano Pis. Maybe this problem is also related to USB OTG somehow?

**Comment by sc7171** [comment]
@ceedveed said:
> I have a different device (ARM based chromebook using crouton), but I get the exact same log. A x86_64 xubuntu laptop works though (need to set verbose boot).

So, these systems **DO NOT** share the same USB controller, as I am running a Raspberry Pi 3 B instead.

**Comment by MJavad** [comment]
@ann0see Can you give me the output of `lsusb` on your Banana Pi M1?
This is what my NanoPi K2 reports **without** any connected devices:
```
# lsusb
Bus 002 Device 002: ID 05e3:0610 Genesys Logic, Inc. 4-port hub
Bus 002 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
```
You can see that there are two root USBs. One for the OTG port and one for the on-board hub.
If the issue is with the hub, the OTG port may actually work as it's connected directly to SoC.
I have purchased an OTG Y cable to attempt this which will arrive today or tomorrow.

**Comment by ann0see** [comment]
That’s my output:
Quite a lot:
`Bus 004 Device 001: ID 1d6b:0001 Linux Foundation 1.1 root hub
Bus 002 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
Bus 003 Device 001: ID 1d6b:0001 Linux Foundation 1.1 root hub
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
Bus 005 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub`

**Comment by FedeltaMedia** [comment]
> Did anybody try e.g. the Giant Board https://www.crowdsupply.com/groboards/giant-board Banana Pi M2 Zero etc?

Well someone here said it did work for the NanoPi Neo2? But I'm unsure. I'd love to see the `lsusb` output for that too

**Comment by ann0see** [comment]
![B8A980C2-571C-45D9-B070-9954383D2DB4](https://user-images.githubusercontent.com/20726856/74109444-e1d73d80-4b83-11ea-8b94-2f4a0c428a8d.jpeg)

That‘s the output of the Pi 3 without anything plugged in. That’s only one controller and a hub probably.

**Comment by sc7171** [comment]
Can we get anyone on the development team to notice this? It seems like the problem isn't being addressed properly.

**Comment by jslay-excella** [comment]
Works on my Pi 4 4GB using the latest Raspbian image (just flashed SD card), 6 Max 12.4.5, 0.9.8 beta arm32. Installed `usbmuxd` to get the `Trust this device` prompt and hit `Trust` as it was still failing for some reason, even in CLI DFU mode. Then put it into DFU mode (I am old school and just start here from my iPhone Edge days, and I get annoyed by the wizard crap)

```
sudo apt update
sudo apt full-upgrade -y
sudo apt autoremove -y
sudo apt autoclean -y
sudo apt install usbmuxd -y
sudo reboot
sudo ./checkra1n -c 
```

```
pi@raspberrypi:~ $ sudo ./checkra1n  -c
#
# Checkra1n beta 0.9.8
#
# Proudly written in nano
# (c) 2019-2020 Kim Jong Cracks
#
#========  Made by  =======
# argp, axi0mx, danyl931, jaywalker, kirb, littlelailo, nitoTV
# never_released, nullpixel, pimskeks, qwertyoruiop, sbingner, siguza
#======== Thanks to =======
# haifisch, jndok, jonseals, xerub, lilstevie, psychotea, sferrini
# Cellebrite (ih8sn0w, cjori, ronyrus et al.)
#==========================

 - [*]: Waiting for DFU devices
 - [*]: Exploiting
 - [*]: Checking if device is ready
 - [*]: Setting up the exploit (this is the heap spray)
 - [*]: Right before trigger (this is the real bug setup)
 - [*]: Entered download mode
 - [*]: All Done
```

```
pi@raspberrypi:~ $ uname -a
Linux raspberrypi 4.19.97-v7l+ #1294 SMP Thu Jan 30 13:21:14 GMT 2020 armv7l GNU/Linux
```

**Comment by jslay88** [comment]
Works on my Pi 4 4GB using the latest Raspbian image (just flashed SD card), 6 Plus, 12.4.5, 0.9.8 beta arm32, plugged into USB A 3.0 Port. Installed `usbmuxd` to get the `Trust this device` prompt and hit `Trust` as it was still failing for some reason, even in CLI DFU mode, this also allows the TUI to work for headless Pi. Then put it into DFU mode (I am old school and just start here from my iPhone Edge days, and I get annoyed by the wizard crap)

I cannot overstate how important it is to ensure you have a proper power supply for your Pi's. This may be part of the issue for some people. Or maybe it really is screwed on Pi 3/Arch. I can test that later. I have one of those as well. 

```
sudo apt update
sudo apt full-upgrade -y
sudo apt autoremove -y
sudo apt autoclean -y
sudo apt install usbmuxd -y
sudo reboot
sudo ./checkra1n -c 
```

```
pi@raspberrypi:~ $ sudo ./checkra1n  -c
#
# Checkra1n beta 0.9.8
#
# Proudly written in nano
# (c) 2019-2020 Kim Jong Cracks
#
#========  Made by  =======
# argp, axi0mx, danyl931, jaywalker, kirb, littlelailo, nitoTV
# never_released, nullpixel, pimskeks, qwertyoruiop, sbingner, siguza
#======== Thanks to =======
# haifisch, jndok, jonseals, xerub, lilstevie, psychotea, sferrini
# Cellebrite (ih8sn0w, cjori, ronyrus et al.)
#==========================

 - [*]: Waiting for DFU devices
 - [*]: Exploiting
 - [*]: Checking if device is ready
 - [*]: Setting up the exploit (this is the heap spray)
 - [*]: Right before trigger (this is the real bug setup)
 - [*]: Entered download mode
 - [*]: All Done
```

```
pi@raspberrypi:~ $ uname -a
Linux raspberrypi 4.19.97-v7l+ #1294 SMP Thu Jan 30 13:21:14 GMT 2020 armv7l GNU/Linux
```

**Comment by MJavad** [comment]
Well, I have bad news.
I tried USB OTG Y cable on NanoPi K2, it didn't work.
```
root@nanopik2:~# lsusb
Bus 002 Device 002: ID 05e3:0610 Genesys Logic, Inc. 4-port hub
Bus 002 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
Bus 001 Device 006: ID 05ac:1227 Apple, Inc. Mobile Device (DFU Mode)
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
```
```
root@nanopik2:~# ./checkra1n -c -v -V
# Checkra1n beta 0.9.8
...
#==========================
- [02/10/20 09:42:48] <Info>: Waiting for DFU devices
- [02/10/20 09:42:48] <Verbose>: DFU device connected: 648a428a2efae
- [02/10/20 09:42:48] <Info>: Exploiting
- [02/10/20 09:42:48] <Verbose>: Attempting to perform checkm8 on 8015 11...
- [02/10/20 09:42:48] <Info>: Checking if device is ready
- [02/10/20 09:42:48] <Verbose>: == Checkm8 Preparation stage ==
- [02/10/20 09:42:50] <Info>: Setting up the exploit (this is the heap spray)
- [02/10/20 09:42:50] <Verbose>: == Checkm8 Setup stage ==
USBDEVFS_SUBMITURB: No such device
USBDEVFS_REAPURB: No such device
- [02/10/20 09:42:59] <Error>: [EXPLOIT:!] Failed issuing transaction abort (error code: -28)
- [02/10/20 09:42:59] <Error>: Exploit failed! (error code: -21)
- [02/10/20 09:42:59] <Error>: Unable to perform checkm8: ffffffff (error code: -21)
```
But on NanoPi Neo2 OTG Y cable works fine.
So it seems the issue is not **just** with the hub, somthing else is wrong too.

@Fidelity88 Yes, it's working on my NanoPi Neo2 from both USB and OTG.
Also, the SoC seems to have other usb controllers which are accessible from GPIO pins.
lsusb output for Neo2 with two connected devices:
```
root@nanopineo2:~# lsusb
Bus 008 Device 001: ID 1d6b:0001 Linux Foundation 1.1 root hub
Bus 005 Device 002: ID 05ac:1227 Apple, Inc. Mobile Device (DFU Mode)
Bus 005 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
Bus 007 Device 001: ID 1d6b:0001 Linux Foundation 1.1 root hub
Bus 004 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
Bus 006 Device 001: ID 1d6b:0001 Linux Foundation 1.1 root hub
Bus 003 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
Bus 002 Device 001: ID 1d6b:0001 Linux Foundation 1.1 root hub
Bus 001 Device 002: ID 05ac:1227 Apple, Inc. Mobile Device (DFU Mode)
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
Bus 009 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
```

**Comment by ann0see** [comment]
So do the USB Pins work on the Nano Pi 2?

**Comment by MJavad** [comment]
@ann0see
> That’s my output:

From this and my tests with OTG, I assume the issue could be presence of on-board hub even if it's not used.
[NanoPi 2](http://wiki.friendlyarm.com/wiki/index.php/NanoPi_2) doesn't seem to have a hub so there is a chance it works.

**Comment by ann0see** [comment]
Hmm. So does the pi zero also have an onboard hub? Because it doesn’t work on it either, I think (not yet tested. Just with a Pi Zero W)

**Comment by Nikostito** [comment]
From my understanding is all due to these devices using weird usb controllers. Nothing to do with the onboard hubs. A guy on discord mad eit work with an orangepi zer0 which just happens to use a different usb controller from what rpi uses. Everything else is practically same. So, it is really defficult for the dev team to fix every single one usb controller out there.

**Comment by sc7171** [comment]
Yeah, but there was a video on YouTube with a raspberry pi 3 Running checkra1n properly I’m pretty sure. https://youtu.be/3ejPqYn1gOY

**Comment by Nikostito** [comment]
> Yeah, but there was a video on YouTube with a raspberry pi 3 Running checkra1n properly I’m pretty sure. https://youtu.be/3ejPqYn1gOY

Had forgotten the existence of this video... Really weird then why pi3 ins't running out of the box.

**Comment by jslay88** [comment]
Is there any way for me to easily test the efficacy of other Pi's/distros without restoring the phone?

**Comment by Nikostito** [comment]
> Is there any way for me to easily test the efficacy of other Pi's/distros without restoring the phone?

Why would you restore your phone? You just need to run checkra1n on each one of the devices you own. No restoring whatsoever. But I can assure you that the jb tool fails on rpi zer0/1/2/3

**Comment by jslay88** [comment]
Its already jailbroken on 12.4.5. I have 1 iOS device that I own that is exploitable via checkm8.  I have multiple Pis from 1-4 + zero, and want to test between all of those as quickly as possible. 

I mean, I figure it would be relevant building up a data matrix of what does and doesnt work.

**Comment by ann0see** [comment]
We could make a google speed sheet. Unfortunately I don’t know how to handle it (and I don’t want to use Google). Any alternatives?
By the way: this Issue is quite off topic now. We should maybe move over to the one I tagged and summarise what has happened been written here.

**Comment by jslay88** [comment]
> We could make a google speed sheet. Unfortunately I don’t know how to handle it (and I don’t want to use Google). Any alternatives?
> By the way: this Issue is quite off topic now. We should maybe move over to the one I tagged and summarise what has happened been written here.

We can make a table within a github comment.

```
| First Header  | Second Header |
| ------------- | ------------- |
| Content Cell  | Content Cell  |
| Content Cell  | Content Cell  |
```

| First Header  | Second Header |
| ------------- | ------------- |
| Content Cell  | Content Cell  |
| Content Cell  | Content Cell  |

**Comment by FedeltaMedia** [comment]
I agree, let's use a Google spreadsheet. Github can do too, but then again, this is a bug tracker. And we're going quite off-topic

**Comment by dim4der** [comment]
Hi! I just came across this issue. Has the spreadsheet already been created? If no, I created a document via Dropbox Paper: https://www.dropbox.com/scl/fi/2k0jm70vtk4eo84wzvzrf/Checkra1n-SBC-compatibility.paper?dl=0&rlkey=xbfiqv38n0slp623s40fu5iwj
It's not yet complete, but I will be looking into it.

**Comment by gilshahar7** [comment]
Just tried to use the checkra1n cli on my rpi 3 B+ and it also failed with similar errors.

Anyone figured out something already or are we waiting for the checkra1n team to further investigate?

**Comment by dim4der** [comment]
Updated the sheet with the PocketBeagle, which is working sometimes (https://github.com/checkra1n/BugTracker/issues/779#issuecomment-589971383) and the BananaPi M2 Zero, which works (thanks to a comment on Dropbox). Can anybody confirm these two boards?

**Comment by wmacintosh** [comment]
Has anyone tried orange Pi?

**Comment by gilshahar7** [comment]
There is someone on reddit claiming to get it working on a raspberry pi zero after he compiled usbmuxd from source.

Here is his comment: https://reddit.com/r/jailbreak/comments/f8nkn9/_/finfhha/?context=1

Can anyone confirm?

**Comment by Nikostito** [comment]
I compiled usbmuxd, libusbmuxd, liblist and libmobiledevice from source and installed them. Still getting the same errror an a pi zer0.

This is the script I used if anyone wanna try it. To run it paste it in a.sh file run sudo chmod +x file.sh and then ./file.sh


```
sudo apt-get update
sudo apt-get -y upgrade
sudo apt-get -y install vim tmux git build-essential libxml2-dev python2.7 python2.7-dev fuse libtool autoconf libusb-1.0-0-dev libfuse-dev

mkdir iphone_libs && cd iphone_libs

git clone https://github.com/libimobiledevice/libplist.git
git clone https://github.com/libimobiledevice/libusbmuxd.git
git clone https://github.com/libimobiledevice/usbmuxd.git
git clone https://github.com/libimobiledevice/libimobiledevice.git
git clone https://github.com/libimobiledevice/ifuse.git

export PKG_CONFIG_PATH=/usr/local/lib/pkgconfig
echo "export PKG_CONFIG_PATH=/usr/local/lib/pkgconfig" | sudo tee -a /etc/bash.bashrc

cd libplist && ./autogen.sh && make && sudo make install && cd ..
cd libusbmuxd && ./autogen.sh && make && sudo make install && cd ..
cd libimobiledevice && ./autogen.sh && make && sudo make install && cd ..
cd usbmuxd && ./autogen.sh && make && sudo make install && cd ..
cd ifuse && ./autogen.sh && make && sudo make install && cd ..

groupadd -g 140 usbmux &>/dev/null
useradd -c 'usbmux user' -u 140 -g usbmux -d / -s /sbin/nologin usbmux &>/dev/null
passwd -l usbmux &>/dev/null

echo /usr/local/lib | sudo tee /etc/ld.so.conf.d/libimobiledevice-libs.conf
sudo ldconfig
```

**Comment by Nikostito** [comment]
> Hi! I just came across this issue. Has the spreadsheet already been created? If no, I created a document via Dropbox Paper: https://www.dropbox.com/scl/fi/tryvbqm15vd71cao7hkgt/Checkra1n-micro-pc-compatibility.paper?dl=0&rlkey=j1p5sua2yvowusjr6ouligb7x
> It's not yet complete, but I will be looking into it.

You should let others with the link edit this paper.

**Comment by Bizzaro** [comment]
I think the raspberry pi zero will never work. The guy that the reddit post used a banana pi zero m2 which is a different thing from the raspberry pi zero. 
I suspect it's because of the USB host hardware config of the raspberry pi zero.

**Comment by gilshahar7** [comment]
Yeah the original poster used a banana pi but the guy that commented said
He did it on a raspberry pi. Not sure if we can believe him though.

**Comment by dim4der** [comment]
> > Hi! I just came across this issue. Has the spreadsheet already been created? If no, I created a document via Dropbox Paper:
https://www.dropbox.com/scl/fi/2k0jm70vtk4eo84wzvzrf/Checkra1n-SBC-compatibility.paper?dl=0&rlkey=xbfiqv38n0slp623s40fu5iwj
> > It's not yet complete, but I will be looking into it.
> 
> You should let others with the link edit this paper.

I sent you a link via dropbox. I hope you received it. Should be on your mail account.

**Comment by dim4der** [comment]
Unfortunately I had to update the link. This is the new one: https://www.dropbox.com/scl/fi/2k0jm70vtk4eo84wzvzrf/Checkra1n-SBC-compatibility.paper?dl=0&rlkey=xbfiqv38n0slp623s40fu5iwj

**Comment by Bizzaro** [comment]
tl;dr recompiling doesn't work on rpi zero.
fresh image of raspbian (2020-02-05-raspbian-buster-lite.img)
```
sudo apt-get update
sudo apt-get install make automake autoconf libtool pkg-config gcc udev libimobiledevice-dev libplist-dev libusb-1.0-0-dev git -y 

sudo apt-get --purge remove usbmuxd

#make sure its gone
usbmuxd --version

git clone https://github.com/libimobiledevice/usbmuxd.git

./autogen.sh
make
sudo make install

#relog to ssh
#should return v 1.1.1 now
usbmuxd --version

Exploit Failed Error code: -21
USBDEVFS_SUBMITURB: No such device
USBDEVFS_REAPURB: No such device
```

That guy's probably running a banana pi as well. There's no way compiling from source worked. The version of usbmuxd from the apt repos is 1.1.1, same as the one he used to compile manually from the source on GitHub.

**Comment by BenjaminHae** [comment]
Maybe it works if you use debian instead of raspbian. But you have to build the image yourself I think:
https://wiki.debian.org/RaspberryPiImages

**Comment by Nikostito** [comment]
I ordered an orangePi zero. Maybe the chapest solution of them all.

**Comment by ann0see** [comment]
> I ordered an orangePi zero. Maybe the chapest solution of them all.

Does the orange pi work?

**Comment by mirsys** [comment]
`- [03/11/20 13:36:14] <Verbose>: DFU device connected: 
 - [03/11/20 13:36:14] <Info>: Exploiting
 - [03/11/20 13:36:14] <Verbose>: Attempting to perform checkm8 on 8015 11...
 - [03/11/20 13:36:14] <Info>: Checking if device is ready
 - [03/11/20 13:36:14] <Verbose>: == Checkm8 Preparation stage ==
 - [03/11/20 13:36:15] <Info>: Setting up the exploit (this is the heap spray)
 - [03/11/20 13:36:15] <Verbose>: == Checkm8 Setup stage ==
 - [03/11/20 13:36:15] <Verbose>: Disabled probabilistic mode since we encounter              ed a partial xfer
 - [03/11/20 13:36:15] <Verbose>: Deterministic approach was successful!
 - [03/11/20 13:36:16] <Info>: Right before trigger (this is the real bug setup)
 - [03/11/20 13:36:16] <Verbose>: Entered initial checkm8 state after 0 steps, i              ssuing DFU abort..
 - [03/11/20 13:36:16] <Verbose>: DFU device connected: 
 - [03/11/20 13:36:16] <Verbose>: == Checkm8 Trigger stage ==
 - [03/11/20 13:36:16] <Verbose>: Checkmate!
 - [03/11/20 13:36:17] <Verbose>: DFU device connected: 
 - [03/11/20 13:36:17] <Verbose>: == Checkm8 Trying to run payload... ==
 - [03/11/20 13:36:18] <Verbose>: If everything went correctly, you should now h              ave code execution.
 - [03/11/20 13:36:22] <Info>: Entered download mode
 - [03/11/20 13:36:22] <Verbose>: Download mode device found
 - [03/11/20 13:36:22] <Info>: Booting...
 - [03/11/20 13:36:22] <Verbose>: Setting bootargs to: rootdev=md0
 - [03/11/20 13:36:48] <Verbose>: Found USBMux interface /sys/devices/platform/s              oc/1c1d000.usb/usb4/4-1/4-1:4.1 with endpoints 85 and 04
 - [03/11/20 13:36:48] <Verbose>: Using ra1nmux
 - [03/11/20 13:36:48] <Verbose>: ra1nmux: connected
 - [03/11/20 13:36:48] <Info>: Uploading bootstrap...
 - [03/11/20 13:36:48] <Verbose>: Uploading bootstrap... 3
 - [03/11/20 13:36:48] <Verbose>: connection closed
 - [03/11/20 13:36:49] <Verbose>: ra1nmux: disconnected
 - [03/11/20 13:36:49] <Info>: All Done
 - [03/11/20 13:36:49] <Verbose>: Bootstrap upload done
` NANOPI NEO2 works fine

**Comment by ape-gg** [comment]
I have achieved it in rpi3 I don't know if it's magic luck oh something but if it works for me..

rpi3 test by me 
rpi4 test by me

**Comment by shrayus-masanam** [comment]
> I have achieved it in rpi3 I don't know if it's magic luck oh something but if it works for me..
> 
> rpi3 test by me
> rpi4 test by me

Can you reproduce what you did to get it to work? What OS on the RPi3?

**Comment by ape-gg** [comment]
> > I have achieved it in rpi3 I don't know if it's magic luck oh something but if it works for me..
> > rpi3 test by me
> > rpi4 test by me
> 
> Can you reproduce what you did to get it to work? What OS on the RPi3?

Only install clean rasbian 

sudo apt-get update
sudo apt-get -y upgrade
sudo apt-get -y install vim tmux git build-essential libxml2-dev python2.7 python2.7-dev fuse libtool autoconf libusb-1.0-0-dev libfuse-dev

**Comment by ape-gg** [comment]
> > > I have achieved it in rpi3 I don't know if it's magic luck oh something but if it works for me..
> > > rpi3 test by me no work— sorry no work 
> > > rpi4 test by me
> >   bpi zero m2 test by me 

> > 
> > Can you reproduce what you did to get it to work? What OS on the RPi3?
> 
> Only install clean rasbian
> 
> sudo apt-get update
> sudo apt-get -y upgrade
> sudo apt-get -y install vim tmux git build-essential libxml2-dev python2.7 python2.7-dev fuse libtool autoconf libusb-1.0-0-dev libfuse-dev

It’s work for my 

Telegram @ape_gg for video and picture

**Comment by shrayus-masanam** [comment]
> > > I have achieved it in rpi3 I don't know if it's magic luck oh something but if it works for me..
> > > rpi3 test by me
> > > rpi4 test by me
> > 
> > 
> > Can you reproduce what you did to get it to work? What OS on the RPi3?
> 
> Only install clean rasbian
> 
> sudo apt-get update
> sudo apt-get -y upgrade
> sudo apt-get -y install vim tmux git build-essential libxml2-dev python2.7 python2.7-dev fuse libtool autoconf libusb-1.0-0-dev libfuse-dev

I'm still stuck at the heap spray using Raspbian Buster Lite and installing all of those packages. Is there a specific Raspbian I need?

**Comment by dim4der** [comment]
> It’s work for my
> 
> Telegram @ape_gg for video and picture

Could you please share it here?

**Comment by Nikostito** [comment]
My OrangePi Zer0 arrived today. It works. I will make a tutorial on reddit on how to Jailbreak your device with a headless pi as a farewell to this community. It was good while it lasted!:)

**Comment by ape-gg** [comment]
![F44B7F03-889A-4858-8ED5-FC5E12EA10C0](https://user-images.githubusercontent.com/62195115/82523498-08e1cb80-9af2-11ea-8e5b-ee13eeecc950.jpeg)
![887CFCB8-C29C-4656-A2A3-AC32E799D3F0](https://user-images.githubusercontent.com/62195115/82523499-097a6200-9af2-11ea-9bb0-612bb1a2af7f.jpeg)
![Uploading E91A65CD-6A48-4A44-B266-A24046941C6A.jpeg…]()

**Comment by ape-gg** [comment]
![AE35B8F8-5673-47B6-8F8D-41C7A089C10A](https://user-images.githubusercontent.com/62195115/82523528-1eef8c00-9af2-11ea-95d4-f10aa75ecd79.jpeg)

**Comment by ape-gg** [comment]
Bpi zero m2 best option w flirc case 👍🏾👍🏾

**Comment by ape-gg** [comment]
![image](https://user-images.githubusercontent.com/62195115/82523732-ab01b380-9af2-11ea-8f47-711ed6b7a712.jpg)
![image](https://user-images.githubusercontent.com/62195115/82523859-f1efa900-9af2-11ea-9500-d49ae17bbb11.jpg)

**Comment by Nikostito** [comment]
Banana is way more expensive though.

**Comment by ape-gg** [comment]
> Banana is way more expensive though.
Price
Bpi zero m2- 24usd
Otg dji- 6dollar
MicroSD- 9dollar
Flirc Case- 19dollar

**Comment by Skye7171** [comment]
At this point I decided to get a RPI 4 4gb since no one ever fixed the issue, and since this ticket has been hijacked and is repeatedly getting necro’d.

**Comment by Skye7171** [comment]
We need someone to close this.

**Comment by Bizzaro** [comment]
im not sure if they will ever fix it, it seems like a hardware limitation.  
it works on old android phones but you might need root though.

**Comment by Skye7171** [comment]
Yes, which is why this ticket isn’t needed anymore. That is the solution of this ticket.

**Comment by dim4der** [comment]
Added the Orange Pi 0 to the Dropbox Paper.
